cleanup_utils/mysql_cleanup.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2019 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+MySQL Cleanup Script
+
+Cleans up all MySQL databases left behind from CI
+"""
+
+import logging
+import re
+from datetime import datetime
+from urllib.parse import urlparse
+
+import MySQLdb as mysql
+from MySQLdb import Error as MySQLError
+
+
+# Logging ####################################################################
+
+logger = logging.getLogger('integration_cleanup')
+
+
+# Functions ##################################################################
+
+def _get_cursor(url):
+    """
+    Returns a cursor on the database
+    """
+    database_url_obj = urlparse(url)
+    try:
+        connection = mysql.connect(
+            host=database_url_obj.hostname,
+            user=database_url_obj.username or '',
+            passwd=database_url_obj.password or '',
+            port=database_url_obj.port or 3306,
+        )
+    except MySQLError as exc:
+        logger.excecption('Cannot get MySQL cursor: %s', exc)
+        raise
+    return connection.cursor()
+
+
+# Classes ####################################################################
+
+class MySqlCleanupInstance:
+    """
+    Handles the cleanup of old MySQL databases
+    """
+    def __init__(self, age_limit, url, domain, dry_run):
+        """
+        Set up variables needed for cleanup
+        """
+        self.age_limit = age_limit
+        self.cleaned_up_hashes = []
+        self.cursor = _get_cursor(url)
```

Comment:
If we pass a URL that is either broke or unaccessible, this will throw an exception. 
Can you add a handler for this to skip cleaning up MySQL if a connection error happens?
I've done something similar [here](https://github.com/open-craft/opencraft/blob/master/cleanup_utils/openstack_cleanup.py#L51).

We want this script to be as error tolerant as possible. This is needed because if this module throws an exception, we will lose all those hashes from the cleanups that ran before this one.